ext/ldap/ldap.c

```diff
@@ -427,12 +429,13 @@ static int _php_ldap_control_from_array(LDAP *ld, LDAPControl** ctrl, zval* arra
 							goto failure;
 						}
 
-						tmpstring = zval_get_string(attr);
+						tmpstrings1[num_tmpstrings1] = zval_get_string(attr);
+						++num_tmpstrings1;
```

Comment:
For clarity, I'd move this to the very end and avoid the -1. zval_get_string returns an interned string in case of exception, so releasing it is optional. Same for the case below.

Reply:
fixed in new commit